ignite/engine/__init__.py

```diff
@@ -91,20 +101,54 @@ def create_supervised_trainer(
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
+    has_native_amp, has_apex_amp = False, False
 
     if on_tpu and not idist.has_xla_support:
         raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
 
+    if amp and LooseVersion(torch.__version__) >= LooseVersion("1.6.0"):
+        from torch.cuda.amp import autocast
+
+        has_native_amp = True
+    elif amp:
+        try:
+            from apex import amp as apex_amp
+
+            has_apex_amp = True
+        except ImportError:
+            raise ImportError("Please install apex from https://github.com/nvidia/apex to use amp.")
+
     def _update(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
         model.train()
         optimizer.zero_grad()
         x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
-        y_pred = model(x)
-        loss = loss_fn(y_pred, y)
-        loss.backward()
+        if has_native_amp:
+            with autocast(enabled=amp):
+                y_pred = model(x)
+                loss = loss_fn(y_pred, y)
+        else:
+            y_pred = model(x)
+            loss = loss_fn(y_pred, y)
+
+        if has_native_amp:
+            scaler.scale(loss).backward()
+            scaler.unscale_(optimizer)
```

Comment:
Let's perform unscale below before the line 145.

Reply:
Any reason why? We need another `if` check.

Reply:
Normally, we do not need to unscale the gradients. So if user do not clip, it is an unnecessary op, right?
However, if user provides grad_norm_kwargs we will unscale and perform the clipping.

Reply:
The docs say even if we don't do `unscale_`, it performs internally when we called ,`step`. So it doesn't relate to grad clip.

Reply:
OK, I see. However, I think it would be more simple to understand the logic as 
```python
scaler.unscale_(optimizer)
torch.nn.utils.clip_grad_norm_(model.parameters(), **grad_norm_kwargs)
```
and avoid user's questions where we just point out to the official docs as you did right now ;)